### PR TITLE
spl-token-cli: support ATA in close, balance, and account-info

### PR DESCRIPTION
**Pull request by CriesofCarrots**
Various spl-token cli subcommands require account addresses be passed in. This is inconsistent with the general emphasis on ATA, and the standard of one ATA per main wallet address per mint.

This PR updates close, balance, and account-info to expect a mint address and use the ATA by default. Closing or querying a specific account is still available using the `--address` parameter.

Also adds an `Aux` tag to account infos of non-ATAs

### CODE REVIEW

File: token/cli/src/main.rs [hunk 1]
```diff
@@ -1703,22 +1713,32 @@ fn main() {
...
-                        .index(2)
                         .help("The address of the account to receive remaining SOL [default: --owner]"),
                 )
+                .arg(
+                    Arg::with_name("address")
+                        .long("address")
+                        .validator(is_valid_pubkey)
+                        .value_name("TOKEN_ACCOUNT_ADDRESS")
+                        .takes_value(true)
+                        .required_unless("token")
```
Comment (t-nelson) [inline token/cli/src/main.rs:1779]: Does this include `--address` in the error message if the `token` positional is omitted?  I think ideally we only suggest the ATA usage and they can `--help` to find `--address`.  I'm not sure if we can trick `clap` into that though?

**Review by t-nelson — APPROVED** [review]
lgtm!  Just one nit about arg specification

File: token/cli/src/main.rs (hunk 1 above)
Comment (CriesofCarrots) [reply token/cli/src/main.rs:1779]: Currently, yes, and I agree it's not ideal. Let me try some asymmetric options to see if I can hide `--address` from that error msg...

Comment (CriesofCarrots) [reply token/cli/src/main.rs]: K, I think this is nicer:

```
$ spl-token balance
error: The following required arguments were not provided:
    <TOKEN_ADDRESS>

USAGE:
    spl-token balance <TOKEN_ADDRESS> --config <PATH>
```

```
$ spl-token balance 2ywp5tbpeeEUAM2PFP6NPtvovyFVnjX3qNfrd13S3HmG --address CYRJWqiSjLitBAcRxPvWpgX3s5TvmN2SuRY3eEYypFvT
error: The argument '<TOKEN_ADDRESS>' cannot be used with '--address <TOKEN_ACCOUNT_ADDRESS>'

USAGE:
    spl-token balance <TOKEN_ADDRESS> --address <TOKEN_ACCOUNT_ADDRESS> --config <PATH>
```

and `spl-token balance <TOKEN_ADDRESS>` and `spl-token balance --address <TOKEN_ACCOUNT_ADDRESS>` just work

Comment (t-nelson) [reply token/cli/src/main.rs]: Lovely!  Thanks for digging in on that!

**Review by t-nelson — APPROVED** [review]
:shipit:

**Pull request merged**